libs/encryption-sdk/src/test/java/org/opensearch/encryption/OffsetRangeFileInputStream.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.encryption;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+public class OffsetRangeFileInputStream extends InputStream implements Closeable {
+    private final InputStream inputStream;
+    private final FileChannel fileChannel;
+    private final long actualSizeToRead;
+    private final long limit;
+    private long counter = 0L;
+    private long markPointer;
+    private long markCounter;
+
+    public OffsetRangeFileInputStream(Path path, long size, long position) throws IOException {
+        this.fileChannel = FileChannel.open(path, StandardOpenOption.READ);
+        this.fileChannel.position(position);
+        this.inputStream = Channels.newInputStream(this.fileChannel);
+        long totalLength = this.fileChannel.size();
+        this.counter = 0L;
+        this.limit = size;
+        if (totalLength - position > this.limit) {
+            this.actualSizeToRead = this.limit;
+        } else {
+            this.actualSizeToRead = totalLength - position;
+        }
+
+    }
+
+    public int read(byte[] b, int off, int len) throws IOException {
```

Comment:
1. Lets add @Override to all interfaces that are being overridden to understand if this class introduces new ones
2. Should this be in commons rather that being places in encryption directory

Reply:
Removed this class as it wasn't being used anymore.